hbase_storage/src/main/java/com/splicemachine/si/data/hbase/coprocessor/SIObserver.java

```diff
@@ -508,4 +510,22 @@ protected boolean aclCheck(Scan scan) throws IOException, StandardException {
             }
         }
     }
+
+    @Override
+    public void preWALRestore(ObserverContext<? extends RegionCoprocessorEnvironment> ctx, RegionInfo info, WALKey logKey, WALEdit logEdit) throws IOException {
+
+        SConfiguration config = HConfiguration.getConfiguration();
+        String namespace = logKey.getTableName().getNamespaceAsString();
+        if (namespace.equals(config.getNamespace())) {
+            ArrayList<Cell> cells = logEdit.getCells();
+            Iterator<Cell> it = cells.iterator();
+            while (it.hasNext()) {
+                Cell cell = it.next();
+                CellType cellType = CellUtils.getKeyValueType(cell);
+                if (cellType == CellType.FIRST_WRITE_TOKEN) {
```

Comment:
So, if I understand correctly, we will remove the `FIRST_WRITE_TOKEN` cell from the list of WAL edits, therefore it will be written to the HFile causing a proper deletion to happen when compaction kicks in, right?

Reply:
+1